apps/system/js/advanced_telemetry.js

```diff
@@ -267,6 +267,7 @@
       // If it's a merge request mark it as completed.
       if (self.merge) {
         self.merge = false;
```

Comment:
So, a different action is taken based on whether it was a merge or a transmit timeout.  If it was a merge timeout, we just need to merge/save to asynch/clear gecko metrics.  If it was a transmit timeout, we need to merge/transmit/delete asynch/clear gecko metrics.

The mergeTimeStart gets set when we ask gecko for the payload.  Are you suggesting to move that to after the mergeMetrics completes?